src/context/FavoritesContext.tsx

```diff
@@ -0,0 +1,40 @@
+import { createContext, useEffect, useState } from 'react';
+import { Product } from '../types/ProductCard';
+
+type FavoritesContextType = {
+  favoriteProducts: Product[];
+  setFavoriteProducts: React.Dispatch<React.SetStateAction<Product[]>>;
+};
+
+export const FavoritesContext = createContext<FavoritesContextType>({
+  favoriteProducts: [],
+  setFavoriteProducts: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const FavoritesProvider: React.FC<Props> = ({ children }) => {
+  const [favoriteProducts, setFavoriteProducts] = useState<Product[]>([]);
+
+  useEffect(() => {
```

Comment:
This would be much easier if only you had a custom hook to work with local storage here, isn't it?.. :)